### DISCUSSION TITLE
Combine next-intl/middleware with another middleware that does rewrites.

### DISCUSSION THREAD

**Discussion by Meemaw** [amannn/next-intl#744] [General]
Hi!

We have a usecase, where we want to segment our app based on 2 personas, e.g `basic` and `advanced` user.

For this we need to add a `[persona]` dynamic segment to our app.

This results in the following structure -> `[locale]/[persona]/...` or `[persona]/[locale]/...`. We do not want to expose the `persona` un URL, so we want to rewrite it in the middleware.

I've tried some things, but couldn't get it to work with `next-intl/middleware` out of the box. Do you know if this pattern could be supported somehow, or should we pull out the code from the `next-intl/middleware` and modify it based on our needs?

**Comment by amannn** [comment]
This seems closely related to https://github.com/amannn/next-intl/issues/653. Some users have implemented custom middlewares and navigation APIs for this use case, so maybe that's an option for you too. I'm currently collecting use cases in #653, not sure yet if `next-intl` can/should include built-in support for such customizations.
